airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -126,38 +127,45 @@ def path(self, **kwargs) -> str:
 
 
 # Incremental Streams
-def chunk_date_range(start_date: DateTime, interval=1) -> Iterable[Mapping[str, any]]:
+def chunk_date_range(start_date: DateTime, interval=pendulum.duration(days=1)) -> Iterable[Period]:
     """
-    Returns a list of the beginning and ending timetsamps of each day between the start date and now.
-    The return value is a list of dicts {'oldest': float, 'latest': float} which can be used directly with the Slack API
+    Yields a list of the beginning and ending timestamps of each day between the start date and now.
+    The return value is a pendulum.period
     """
-    intervals = []
     now = pendulum.now()
     # Each stream_slice contains the beginning and ending timestamp for a 24 hour period
     while start_date <= now:
-        end = start_date.add(days=interval)
-        intervals.append({"oldest": start_date.timestamp(), "latest": end.timestamp()})
-        start_date = start_date.add(days=1)
-
-    return intervals
+        end_date = start_date + interval
+        yield pendulum.period(start_date, end_date)
+        start_date = end_date
 
 
 class IncrementalMessageStream(SlackStream, ABC):
     data_field = "messages"
     cursor_field = "ts"
+    primary_key = "id"
```

Comment:
no need to add a new `id` field, we can just refer to `ts` directly

Also the primary key should be `['channel_id', 'ts']` because `ts` is only guaranteed to be unique within a channel (we should add the channel_id in parse_response using the stream slice)

Reply:
Done